segmentation_labeling_app/rois/rois.py

```diff
@@ -0,0 +1,240 @@
+            optional_array: an optional array to overload the function and
+            create edge points from different mask
+
+        Returns:
+            mask_matrix: a binary mask of the edges given a threshold
+
+        """
+        edge_coordinates = self.get_edge_points(threshold=threshold,
+                                                optional_array=optional_array)
+        mask_matrix = np.zeros(shape=self.image_shape)
+        for edge_coordinate in edge_coordinates:
+            mask_matrix[edge_coordinate[1]][edge_coordinate[0]] = 1
+        return mask_matrix
+
+    def create_manifest_json(self, source_ref: str,
+                             video_source_ref: str, max_source_ref: str,
+                             avg_source_ref: str, trace_source_ref: str,
+                             roi_data_source_ref: str) -> str:
+        """
+        Function to make the manifest json string required for sagemaker
+        ground truth to read the data in successfully. The strings are
+        local URIs for each of the required data.
+        Args:
+            source_ref: Location of ROI mask png
+            video_source_ref: Location of 2P video for the ROI
+            max_source_ref: Location of the maximum projection for the ROI
+            avg_source_ref: Location of the average projection for the ROI
+            trace_source_ref: Location of the trace data for the ROI
+            roi_data_source_ref: Location of the ROI coordinate data
+
+        Returns:
+            dictionary: returns dictionary as json string
+        """
+        dictionary = {'source_ref': source_ref,
+                      'video_source_ref': video_source_ref,
+                      'max_source_ref': max_source_ref,
+                      'avg_source_ref': avg_source_ref,
+                      'trace_source_ref': trace_source_ref,
+                      'roi_data_source-ref': roi_data_source_ref,
+                      'roi_id': self.roi_id,
+                      'experiment_id': self.experiment_id}
+        return json.dumps(dictionary)
+
+    def write_roi_to_db(self, database_file_path: Path,
+                        transform_hash: str,
+                        ophys_segmentation_commit_hash: str,
+                        creation_date: str,
+                        manifest: str,
+                        upload_date: str = None):
+        # connect to the database file
+        db_connection = sqlite3.connect(database_file_path.as_posix())
+
+        # create table, sql query handles the checking if already exists
+        curr = db_connection.cursor()
+        curr.execute(sql_create_rois_table)
+
+        roi_task = (transform_hash, ophys_segmentation_commit_hash,
+                    creation_date, upload_date, manifest,
+                    self.experiment_id, self.roi_id)
+        # add the roi to the table
+        curr.execute(sql_insert_roi, roi_task)
+        unique_id = curr.lastrowid
+        db_connection.commit()
```

Comment:
It is probably a good habit to include `db_connection.close()`.I suspect for sqlite, this is not too much of an issue, but I've run into issues in other systems at scale when forgetting these closing calls.